src/hypothesis/internal/reflection.py

```diff
@@ -238,6 +238,8 @@ def args_for_lambda_ast(l):
 SPACE_FOLLOWS_OPEN_BRACKET = re.compile(r"\( ")
 SPACE_PRECEDES_CLOSE_BRACKET = re.compile(r" \)")
 
+LAMBDA_INSIDE_DECORATOR = re.compile(r"@\w+(\s*\(lambda [^\)]+\))")
```

Comment:
A brief comment or breakdown of this regex would be helpful. I think it’s something like:

```
@\w+                 # name of decorator
(\s*                 # start of group, whitespace before decorator brackets?
\(lambda [^\)]+\)    # Lambda body
)                    # end of group
```

but clarification would be helpful.

Reply:
If space is allowed between the decorator and the parens, could we get a test for it?

Ditto space inside the parens – for example, would

```python
@arg_decorator( lambda x: 2 * x )
```

be formatted correctly?

Reply:
Good shout. Trying to clean up and test this regex better made me realise that the logic was badly wrong in a way that made this pull request actually useless.

I've now replaced it with not a regex and added a whole bunch more tests.